### PR TITLE
Add mission creation script

### PR DESCRIPTION
**Pull request by bighomie6384**
Missions can only involve talking to npcs for now.

### CODE REVIEW

**Comment by bighomie6384** [comment]
All mission features are implemented

File: bin/addmission.py [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+			for i in range(min(3, len(taskInfo['failItems']['items']))):
+				mission['m_iFItemID'][i] = taskInfo['failItems']['items'][i]
+				mission['m_iFItemNumNeeded'][i] = taskInfo['failItems']['count'][i]
+
+		for i in range(min(2, len(MISSION_PREREQS))):
+			mission['m_iCSTReqMission'][i] = MISSION_PREREQS[i]
+
+		journalId = createJournalEntry(taskInfo['taskDesc'])
+		mission['m_iSTJournalIDAdd'] = journalId
+		mission['m_iSUJournalIDAdd'] = journalId # NOTE in some missions these are different? but it's a pretty useless thing
```
Comment (kamilprzyb2) [inline bin/addmission.py]:
```suggestion
		mission['m_iSUJournaliDAdd'] = journalId # NOTE in some missions these are different? but it's a pretty useless thing

```
it's actually like that in tabledata /sigh

File: bin/addmission.py [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+
+		mission['m_iHMissionType'] = MISSION_TYPE
+		mission['m_iHDifficultyType'] = MISSION_DIFFICULTY
+		mission['m_iCTRReqLvMin'] = REQUIRED_LEVEL
+
+		mission['m_iHTaskType'] = taskInfo['taskType']
+		mission['m_iHCurrentObjective'] = createMissionString(taskInfo['objective'])
+
+		if 'initialMessage' in taskInfo:
+			mission['m_iSTMessageType'] = 2
```
Comment (kamilprzyb2) [inline bin/addmission.py]:
```suggestion
			mission['m_iSTMessageType'] = taskInfo['initialMessage']['type']

```

File: bin/addmission.py [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+			mission['m_iSTMessageTextID'] = createMissionString(taskInfo['initialMessage']['text'])
+			mission['m_iSTMessageSendNPC'] = taskInfo['initialMessage']['npc']
+
+		if 'initialDialog' in taskInfo:
+			mission['m_iSTDialogBubble'] = createMissionString(taskInfo['initialDialog']['text'])
+			mission['m_iSTDialogBubbleNPCID'] = taskInfo['initialDialog']['npc']
+		
+		if 'successMessage' in taskInfo:
+			mission['m_iSUMessageType'] = 2
```
Comment (kamilprzyb2) [inline bin/addmission.py]:
```suggestion
			mission['m_iSUMessageType'] = taskInfo['successMessage']['type']

```

File: bin/addmission.py [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+
+# Task data
+# Task types - 1: talk to npc 2: touch waypoint 3: interact with object 4: deliver item to npc 5: kill mobs 6: escort
+TASK_DATA = [
+	{
+		'journalNPC': 776,
+		'objective': 'Go to Pokey Oaks Mall.',
+		'taskDesc': 'Investigate the area in Pokey Oaks and see if you can get a clue of what\'s going on.',
+		'initialMessage': {
+			'type': 2,
```
Comment (kamilprzyb2) [inline bin/addmission.py]:
```suggestion
			'type': 2, # Message types - 2: popup 4: email 6: both

```

**Review by kamilprzyb2 — COMMENTED** [review]
message types can be used to send emails

**Review by CakeLancelot — APPROVED** [review]

**Comment by CakeLancelot** [comment]
Any reason why this shouldn't be merged?

**Pull request merged**